Sources/Widgets/Core/WidgetManager/index.js

```diff
@@ -288,38 +298,18 @@ function vtkWidgetManager(publicAPI, model) {
     }
   }
 
-  const handleEvent = (eventName) => {
-    let guard = false;
-    return (callData) => {
-      if (
-        guard ||
-        model.isAnimating ||
-        !model.pickingEnabled ||
-        model._selectionInProgress
-      ) {
-        return macro.VOID;
-      }
-
-      const updatePromise = updateSelection(callData);
-      macro.measurePromiseExecution(updatePromise, (elapsed) => {
-        // 100ms is deemed fast enough. Anything higher can degrade usability.
-        if (elapsed > 100) {
-          macro.vtkWarningMacro(
-            `vtkWidgetManager updateSelection() took ${elapsed}ms on ${eventName}`
-          );
-        }
-      });
+  const handleEvent = async (callData, fromTouchEvent = false) => {
```

Comment:
shoudln't you also support `async` functions in the callback manager ?

Reply:
This is something we'll need to discuss further. Supporting serialization of async event listeners will change the behavior of `invokeXXX` calls (e.g. `obj.modified()`). We either need to ensure that usages of invoke/modified do not assume that all callbacks have run to completion (since async tasks could finish later), or annotate invoke/modified to be async as well.

Reply:
I'm pretty sure that if our event handling system supported serialization of async event handlers, it would eliminate the semi-hacky OnLeftButtonRelease call.

Maybe we can consider an `await invokeAndWait()` concept for instances where you really do want to wait for all callbacks to have been processed before continuing.

Reply:
This will have to be something we explore further as necessary. I won't cover it in the scope of this PR for now.